### ISSUE TITLE
Problems getting sharp library running

### ISSUE THREAD

**Issue by stefanholzapfel** [Urigo/Ionic2CLI-Meteor-WhatsApp#87]
I have trouble installing sharp with meteor's npm:

`meteor npm install --save sharp`

leads to this error:
![image](https://cloud.githubusercontent.com/assets/9963536/24968949/5781a2c2-1faf-11e7-9fe2-ced76081a4d5.png)

I guess the problem is, Meteor uses 32Bit NodeJs?

**Comment by IvanLjubicic** [comment]
Yes, it is. I had the same problem. 
At first, I installed older version of sharp which has 32bit. 
After some difficulties I experienced in development, I decided to "turn off" sharp functionalities in app while I'm working on Windows. Later I'm planning to build real version on other OS.

**Comment by mmazloum** [comment]
I had the similiar problem trying to deploy it on servers on galaxy.meteor, I turned off sharp as well, you can generate thumbs or lower resolution images at upload with gm and it seems to work if you really need this feature.

**Comment by puba3r** [comment]
We had a problem with sharp as well after building the meteor server (meteor build path --architecture XXX). After some digging the issue was the wrong version of sharp being used after building and we fixed it by doing the following steps:

- build and deploy the meteor app (server)
- removing the sharp package from the node_modules
- then doing npm install
- adding the meteor version of sharp: meteor npm install sharp
- starting the server

Those steps are from the back of my head but you should get the idea what to do when you face this issue during deployment.

**Comment by darkbasic** [comment]
This has probably nothing to do with your issue, but since I encoutered this problem and I had to spend quite some time to fix it I will post what I discovered anyway. Sharp bundles a binary library, which doesn't always link fine on your system depending on your libraries versions. At least on Arch, it doesn't. So I had to compile and install such library and it magically started working. I hope this will save someone else some time.
